math/src/field/f64/mod.rs

```diff
@@ -572,3 +532,42 @@ fn exp_acc<const N: usize>(base: BaseElement, tail: BaseElement) -> BaseElement
     }
     result * tail
 }
+
+/// Montgomery reduction (variable time)
+#[inline(always)]
+pub const fn mont_red_var(x: u128) -> u64 {
+    const NPRIME: u64 = 4294967297;
+    let q = (((x as u64) as u128) * (NPRIME as u128)) as u64;
+    let m = (q as u128) * (M as u128);
+    let y = (((x as i128).wrapping_sub(m as i128)) >> 64) as i64;
+    if x < m {
+        return (y + (M as i64)) as u64;
+    } else {
+        return y as u64;
+    };
+}
+
+/// Montgomery reduction (constant time)
+#[inline(always)]
+pub const fn mont_red_cst(x: u128) -> u64 {
+    // See reference above for a description of the following implementation.
+    let xl = x as u64;
+    let xh = (x >> 64) as u64;
+    let (a, e) = xl.overflowing_add(xl << 32);
+
+    let b = a.wrapping_sub(a >> 32).wrapping_sub(e as u64);
+
+    let (r, c) = xh.overflowing_sub(b);
+    r.wrapping_sub(0u32.wrapping_sub(c as u32) as u64)
+}
+
+/// Test of equality between two BaseField elements; return value is
+/// 0xFFFFFFFFFFFFFFFF if the two values are equal, or 0 otherwise.
+#[inline(always)]
+pub fn equals(lhs: u64, rhs: u64) -> u64 {
+    // Since internal representation is canonical, we can simply
+    // do a xor between the two operands, and then use the same
+    // expression as iszero().
```

Comment:
I think this comment needs to be updated (i.e., I don't think we have `iszero()` function).